test/serverless-v2/Relay.ts

```diff
@@ -1222,6 +1222,829 @@ describe("Relay - Job sent by UserSample contract", function () {
+		await expect(tx).to.revertedWithCustomError(relay, "RelayNotJobSubscriptionOwner");
+	});
+
+	it("can update job termination params", async function () {
+		let jobSubsId: any = await relay.jobSubsCount(),
+			terminationTimestamp = await time.latest() + 110,
+			usdcDeposit = 2000000n;
+		
+		let jobSubsInitial = await relay.jobSubscriptions(jobSubsId);
+		
+		let tx = relay.connect(signers[2])
+			.updateJobTerminationParams(jobSubsId, terminationTimestamp, usdcDeposit, 
+				{ value: callbackDeposit }
+			);
+		await expect(tx).to.emit(relay, "JobSubscriptionTerminationParamsUpdated")
+			.withArgs(jobSubsId, terminationTimestamp);
+
+		let jobSubsFinal = await relay.jobSubscriptions(jobSubsId);
+		expect(jobSubsFinal.terminationTimestamp).to.eq(terminationTimestamp);
+		expect(jobSubsFinal.job.usdcDeposit).to.eq(jobSubsInitial.job.usdcDeposit + usdcDeposit);
+		expect(jobSubsFinal.job.callbackDeposit).to.eq(jobSubsInitial.job.callbackDeposit + callbackDeposit);
+	});
+
+	it("cannot update job termination params without job subscription owner account", async function () {
+		let jobSubsId: any = await relay.jobSubsCount(),
+			terminationTimestamp = await time.latest() + 110,
+			usdcDeposit = 2000000;
+
+		let tx = relay.connect(signers[1])
+			.updateJobTerminationParams(jobSubsId, terminationTimestamp, usdcDeposit, 
+				{ value: callbackDeposit }
+			);
+		await expect(tx).to.be.revertedWithCustomError(relay, "RelayNotJobSubscriptionOwner");
+	});
+
+	it("cannot update job termination params with invalid termination timestamp", async function () {
+		let jobSubsId: any = await relay.jobSubsCount(),
+			terminationTimestamp = await time.latest() + 90,
+			usdcDeposit = 2000000;
+
+		let tx = relay.connect(signers[2])
+			.updateJobTerminationParams(jobSubsId, terminationTimestamp, usdcDeposit, 
+				{ value: callbackDeposit }
+			);
+		await expect(tx).to.be.revertedWithCustomError(relay, "RelayInvalidTerminationTimestamp");
+	});
+
+	it("cannot update job termination params with insufficient callback deposit", async function () {
+		let jobSubsId: any = await relay.jobSubsCount(),
+			terminationTimestamp = await time.latest() + 1000,
+			usdcDeposit = 2000000;
+
+		let tx = relay.connect(signers[2])
+			.updateJobTerminationParams(jobSubsId, terminationTimestamp, usdcDeposit, 
+				{ value: 0 }
+			);
+		await expect(tx).to.be.revertedWithCustomError(relay, "RelayInsufficientCallbackDeposit");
+	});
+
+	it("cannot update job termination params with insufficient USDC deposit", async function () {
+		let jobSubsId: any = await relay.jobSubsCount(),
+			terminationTimestamp = await time.latest() + 1000,
+			usdcDeposit = 2000000;
+
+		let tx = relay.connect(signers[2])
+			.updateJobTerminationParams(jobSubsId, terminationTimestamp, usdcDeposit, 
+				{ value: parseUnits("1") }
+			);
+		await expect(tx).to.be.revertedWithCustomError(relay, "RelayInsufficientUsdcDeposit");
+	});
+});
```

Comment:
Also, add a case where new termination time stamp is before the orignal time stamp, you may increase the original timestamp to 1000 instead of 100 (which is clashing with overall timeout).

Reply:
isn't this the same case as the previous one - `it("cannot update job termination params with invalid termination timestamp"`
txn will revert with `RelayInvalidTerminationTimestamp` error

Reply:
no this is a successful case where you are decreasing the terminationTimestamp but above the lower limit.